lib/Console/Application.php

```diff
@@ -64,31 +65,42 @@ public function __construct(KernelInterface $kernel)
 
         $this->setDispatcher($dispatcher);
 
-        $dispatcher->addListener(ConsoleEvents::COMMAND, function (ConsoleCommandEvent $event) use ($kernel) {
-            // skip if maintenance mode is on and the flag is not set
-            if (Admin::isInMaintenanceMode() && !$event->getInput()->getOption('ignore-maintenance-mode')) {
-                throw new \RuntimeException('In maintenance mode - set the flag --ignore-maintenance-mode to force execution!');
-            }
-
-            if ($event->getInput()->getOption('maintenance-mode')) {
-                // enable maintenance mode if requested
-                $maintenanceModeId = 'cache-warming-dummy-session-id';
-
-                $event->getOutput()->writeln('Activating maintenance mode with ID <comment>' . $maintenanceModeId . '</comment> ...');
-
-                Admin::activateMaintenanceMode($maintenanceModeId);
-            }
-
-            if ($event->getCommand() instanceof DoctrineCommand && $prefix = $event->getInput()->getOption('prefix')) {
-                $kernel->getContainer()->get(FilteredMigrationsRepository::class)->setPrefix($prefix);
-                $kernel->getContainer()->get(FilteredTableMetadataStorage::class)->setPrefix($prefix);
-            }
+        $maintenanceModeHelper = $kernel->getContainer()->get(MaintenanceModeHelperInterface::class);
+        $dispatcher->addListener(ConsoleEvents::COMMAND, function (ConsoleCommandEvent $event)
+            use ($kernel, $maintenanceModeHelper) {
+                // skip if maintenance mode is on and the flag is not set
+                if ($maintenanceModeHelper->isActive() || Admin::isInMaintenanceMode() &&
+                    !$event->getInput()->getOption('ignore-maintenance-mode')
```

Comment:
```suggestion
                if (($maintenanceModeHelper->isActive() || Admin::isInMaintenanceMode()) &&
                    !$event->getInput()->getOption('ignore-maintenance-mode')
```
similarly to https://github.com/pimcore/pimcore/pull/15954/files?w=1#diff-ae4fd3053efaec0fe50cc39c28378027e8884af72d1facbb19980ade26f36220R78